File: sorting/InsertionSort/cpp/InsertionSort.cpp
#include "InsertionSort.hpp"

using namespace std;

void insertionSort(vector<long long>& values) {
    for (size_t i = 1; i < values.size(); ++i) {
        const long long key = values[i];
        size_t j = i;

        while (j > 0 && values[j - 1] > key) {
            values[j] = values[j - 1];
            --j;
        }

        values[j] = key;
    }
}

File: sorting/InsertionSort/rust/src/insertion_sort.rs
pub fn insertion_sort(values: &mut [i64]) {
    for i in 1..values.len() {
        let key = values[i];
        let mut j = i;

        while j > 0 && values[j - 1] > key {
            values[j] = values[j - 1];
            j -= 1;
        }

        values[j] = key;
    }
}
